File: 94-BinaryTreeInorderTraversal/94-BinaryTreeInorderTraversal.java
// Last updated: 7/12/2026, 6:23:36 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {

        List<Integer> list = new ArrayList<>();

        Stack<TreeNode> stack = new Stack<TreeNode>();

        TreeNode currNode = root;

        while(!stack.isEmpty() || currNode != null){

            while(currNode != null ){
                stack.add(currNode);
                currNode = currNode.left;
            }

            currNode = stack.pop();
            list.add(currNode.val);
            currNode = currNode.right;
        }

        return list;


        
    }
}

File: 94-BinaryTreeInorderTraversal/94-BinaryTreeInorderTraversal.py
# Last updated: 7/12/2026, 6:23:28 PM
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        st = []
        inorderList = []

        node = root

        while True:
            if node:
                st.append(node)
                node = node.left
            
            else:
                if not st: break

                node = st.pop()
                inorderList.append(node.val)
                node = node.right
            

        return inorderList



                
                


        